packages/instrumentation-langchain/test/instrumentation.test.ts

```diff
@@ -0,0 +1,162 @@
+import {
+  BasicTracerProvider,
+  InMemorySpanExporter,
+  SimpleSpanProcessor,
+} from "@opentelemetry/sdk-trace-base";
+
+import type * as ToolsModule from "langchain/tools";
+import type * as AgentsModule from "langchain/agents";
+import type * as ChainsModule from "langchain/chains";
+import { ChatPromptTemplate, PromptTemplate } from "@langchain/core/prompts";
+import { createOpenAIToolsAgent } from "langchain/agents";
+import { Calculator } from "langchain/tools/calculator";
+import { pull } from "langchain/hub";
+import { HNSWLib } from "@langchain/community/vectorstores/hnswlib";
+import { ChatOpenAI, OpenAI, OpenAIEmbeddings } from "@langchain/openai";
+import { RecursiveCharacterTextSplitter } from "langchain/text_splitter";
+
+import { LangChainInstrumentation } from "../src/instrumentation";
+
+const memoryExporter = new InMemorySpanExporter();
+
+describe("Test LlamaIndex instrumentation", () => {
+  const provider = new BasicTracerProvider();
+  let instrumentation: LangChainInstrumentation;
+  let contextManager: AsyncHooksContextManager;
+  let langchainAgentsModule: typeof AgentsModule;
+  let langchainToolsModule: typeof ToolsModule;
+  let langchainChainsModule: typeof ChainsModule;
+
+  before(() => {
+    provider.addSpanProcessor(new SimpleSpanProcessor(memoryExporter));
+    instrumentation = new LangChainInstrumentation();
+    instrumentation.setTracerProvider(provider);
+    langchainAgentsModule = require("langchain/agents");
+    langchainToolsModule = require("langchain/tools");
+    langchainChainsModule = require("langchain/chains");
+  });
+
+  beforeEach(() => {
+    contextManager = new AsyncHooksContextManager().enable();
+    context.setGlobalContextManager(contextManager);
+  });
+
+  afterEach(() => {
+    memoryExporter.reset();
+    context.disable();
+  });
+
+  it("should set attributes in span for tools and agent instrumentation", async () => {
+    const llm = new ChatOpenAI({});
+    const tools = [new Calculator(), new langchainToolsModule.SerpAPI()];
+    const prompt = await pull<ChatPromptTemplate>(
+      "hwchase17/openai-tools-agent",
+    );
+    const agent = await createOpenAIToolsAgent({ llm, tools, prompt });
+    const agentExecutor = new langchainAgentsModule.AgentExecutor({
+      agent,
+      tools,
+    });
+    const result = await agentExecutor.invoke({
+      input:
+        "By searching the Internet, find how many albums has Boldy James dropped since 2010 and how many albums has Nas dropped since 2010? Find who dropped more albums and show the difference in percent.",
+    });
+
+    const spans = memoryExporter.getFinishedSpans();
+    const spanNames = spans.map((span) => span.name);
+
+    assert.ok(result);
+    assert.ok(spanNames.includes("langchain.task.SerpAPI"));
+    assert.ok(spanNames.includes("langchain.agent"));
+  }).timeout(60000);
+
+  it("should set attributes in span for chain instrumentation", async () => {
```

Comment:
Can you add a test that uses [Langchain LCEL](https://js.langchain.com/docs/expression_language/get_started)?

Reply:
Followed this [doc](https://js.langchain.com/docs/expression_language/cookbook/tools) to add LCEL test. Let me know if I should cover anything else as well.